src/lib/db.ts

```diff
@@ -69,21 +92,20 @@ export function addReaction(
   data: Omit<Reaction, "createdAt" | "rkey" | "did">
 ) {
   const { did, rkey } = parseUri(data.url);
-  const post = db
-    .prepare(`SELECT 1 FROM post WHERE did = ? AND rkey = ?`)
-    .get(did, rkey) as Post | undefined;
-
-  if (!post) {
-    return;
-  }
-
   const id = `${data.id}-${data.type}-${new Date().getTime()}`;
   const result = db
     .prepare(
-      `INSERT OR IGNORE INTO reaction (id, type, did, rkey) VALUES (?, ?, ?, ?)`
+      `
+      INSERT OR IGNORE INTO reaction (id, type, did, rkey)
+      SELECT ?, ?, ?, ?
```

Comment:
This is some high lvl sql haha

So does this basically return the value when `WHERE EXISTS` is true?

Reply:
With the `WHERE` clause it prevents the preceding query if it fails, and in this case we just return `1` if we find a post with the correct `did` and `rkey` but nothing otherwise. The `SELECT ?...` bit is how you can use another table to populate an `INSERT` or `UPDATE` or really anything.

This is a bit new to me, but I felt this was better than doing a separate query to check for a post.